core/commit.go

```diff
@@ -142,6 +150,15 @@ func makeCommitmentCounter(f uint32) commitmentCounter {
 		lock.Lock()
 		defer lock.Unlock()
 
+		if prepareView < view {
+			return false, nil
+		} else if prepareView > view {
+			view = prepareView
+			firstCV = prepareCV
+			lastDoneCV = 0
```

Comment:
I think that `lastDoneCV` should be turned back to the last checkpoint value, but checkpoint is not implemented yet so setting to 0 now. Maybe that's fine.

Reply:
The only purpose of `lastDoneCV` is to limit `prepareStates` in size. In fact, `lastDoneCV` is the highest counter value that is known to have reached the quorum. Zero is not a valid counter value, so it indicates that no quorum has been reached in the view.

Reply:
Okay, thanks for clarification.